### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by ededovets**

### CODE REVIEW

File: SlideController/Classes/SlideContentView.swift [hunk 1]
```diff
@@ -82,52 +85,48 @@ extension ViewSlidableImplementation: ViewSlidable {
...
     
     func removeViewAtIndex(index: Int) {
-        guard index < pages.count else { return }
-        let page: SlideContainerView = pages[index]
-        let prevPage: SlideContainerView? = index > 0 ? pages[index - 1]: nil
-        let nextPage: SlideContainerView? = index < pages.count - 1 ? pages[index + 1]: nil
-        
-        pages.remove(at: index)
+        guard index < containers.count else { return }
+        let page: SlideContainerView = containers[index]
```
Comment (oroom) [inline SlideController/Classes/SlideContentView.swift]: `let page = containers[index]` No need to specify type

File: SlideController/Classes/SlideContentView.swift [hunk 2]
```diff
@@ -82,52 +85,48 @@ extension ViewSlidableImplementation: ViewSlidable {
...
+        let prevPage: SlideContainerView? = index > 0 ? containers[index - 1]: nil
+        let nextPage: SlideContainerView? = index < containers.count - 1 ? containers[index + 1]: nil
+        containers.remove(at: index)
         page.view.removeFromSuperview()
         if let nextPage = nextPage {
-            updateConstraints(nextPage, prevPage: prevPage, isLast: index == pages.count - 1, direction: scrollDirection)
+            updateConstraints(nextPage, prevPage: prevPage, isLast: index == containers.count - 1, direction: slideDirection)
```
Comment (oroom) [inline SlideController/Classes/SlideContentView.swift]: `isLast` will be always false if nextPage is not nil.

File: SlideController/Classes/SlideController.swift [hunk 3]
```diff
@@ -87,30 +76,29 @@ public typealias TitleItemControllableObject = ItemViewable & Initializable & Se
...
 
 public class SlideController<T, N>: NSObject, UIScrollViewDelegate, ControllerSlidable, Viewable where T: ViewSlidable, T: UIScrollView, T: TitleConfigurable, N: TitleItemControllableObject, N: UIView, N.Item == T.View {
+    private let containerView = SlideView<T>()
+    private var titleSlidableController: TitleSlidableController<T, N>!
+    private var slideDirection: SlideDirection!
+    private var contentSlidableController: SlideContentController!
+    private var currentIndex = 0
+    private var lastContentOffset: CGFloat = 0
+    private var didFinishForceSlide: (() -> ())?
+    private var isForcedToSlide: Bool = false
```
Comment (oroom) [inline SlideController/Classes/SlideController.swift]: `: Bool` not needed

File: SlideController/Classes/SlideController.swift [hunk 4]
```diff
@@ -87,30 +76,29 @@ public typealias TitleItemControllableObject = ItemViewable & Initializable & Se
...
+    private var isOnScreen = false
+    private var scrollInProgress: Bool = false
```
Comment (oroom) [inline SlideController/Classes/SlideController.swift]: `: Bool` not needed

File: SlideController/Classes/SlideContentView.swift (hunk 2 above)
Comment (ededovets) [reply SlideController/Classes/SlideContentView.swift]: not for insertView case. @oroom you can refactor this yourself, but fix requires some thinking time :)

**Comment by ededovets** [comment]
@oroom  ready for next review

**Pull request merged**